app/models/framework/miso_fields.rb

```diff
@@ -0,0 +1,46 @@
+require 'csv'
+
+class Framework
+  class MisoFields
+    attr_reader :framework_short_name
+    def initialize(framework_short_name)
+      @framework_short_name = framework_short_name
+    end
+
+    def filename
+      @filename ||= Rails.root.join('data', 'framework_miso_fields.csv')
+    end
+
+    def framework_name
+      for_framework.first['Name']
+    end
+
+    def for_framework
+      @for_framework ||= begin
+        @for_framework = source_csv
```

Comment:
I found this assignment to the same instance variable as the one being set by the conditional assignment a bit surprising and had to work a bit harder to follow what was going on. I also wonder if there is a more appropriate name for this reader than `for_framework`. `framework_fields` perhaps?

Finally  it feels like this is actually an internal implementation detail of the class, with the `invoice_fields` and `order_fields` being the main public interfaces. the tests for those methods should provide all the coverage we need in terms of mapping the csv to the expected fields. then perhaps if this reading/selecting/sorting was moved up into the initialiser we could still have spec coverage for the error being raised for a non-existent framework without having to expose and explicitly test this?

Reply:
Yes that second thing's just a clumsy error. Will correct.  As for the naming it's so we don't have to say `fields` twice at the call site, which will generally look like `miso_fields.for_framework` as opposed to `miso_fields.framework_fields`, which is what it was called before. Also it hammers home the fact that you're really only using this class on a per-framework basis. I think your other comment is correct, though, that this should not be part of the public interface.

Reply:
Actually all of the above :100: – rebased